variables.tf

```diff
@@ -206,14 +206,20 @@ variable "certificate_arn" {
 }
 
 variable "acm_certificate_domain_name" {
-  description = "Route53 domain name to use for ACM certificate. Route53 zone for this domain should be created in advance. Specify if it is different from value in `route53_zone_name`"
+  description = "Route53 domain name to use for ACM certificate. Route53 zone for this domain should be created in advance. Specify if it is different from value in `route53_public_zone_name`"
   type        = string
   default     = ""
 }
 
 # Route53
-variable "route53_zone_name" {
-  description = "Route53 zone name to create ACM certificate in and main A-record, without trailing dot"
+variable "route53_public_zone_name" {
```

Comment:
We can't simply rename existing variables because there are users who are using them.

Instead, let's keep the same variable names as before but add a new boolean/string variable(s) that will be responsible for what you are trying to solve in this PR.

Reply:
I suggested this refactor because the current variable (`route53_private_zone: bool`) for private_zone usage are partially implemented, if the user just enable it, the ACM validation hangs until the Terraform timeouts (You cannot validate a ACM for private hosted zones), to proceed, the user needs to disable certificate ACM generation.
This refactor aims to fix this by creating a single flow for private hosted zone usage, I will revert the variable renaming.